### PR TITLE
Add .hip extension.

### PR DESCRIPTION
**Pull request by breyerml**
Add the .hip extensions for the HIP language dialect (similar to .cu in CUDA).

### CODE REVIEW

**Comment by ClausKlein** [comment]
Note, there is more to do: https://github.com/TheLartians/Format.cmake/issues/41

**Comment by breyerml** [comment]
> Note, there is more to do: #41

I've added the extensions mentioned in this issue. 

Question: is there a mechanism to specify the extensions based on the used clang-format version? Or should I add them unconditional like I did?

File: git-clang-format.py [hunk 1]
```diff
@@ -89,7 +89,6 @@ def main():
       'js',  # JavaScript
       'ts',  # TypeScript
       'cs',  # C Sharp
-      'json',  # Json
```
Comment (ClausKlein) [inline git-clang-format.py]: I use this to format the `cmake presets` and other json files!

Comment (breyerml) [reply git-clang-format.py]: What clang-format version are you using? I tried it with clang-format 18.0.0 (`clang-format version 18.0.0 (/mnt/agent/work/llvm-project/llvm-project/clang 367a7b9b0f9ac6e2a97cdfda4d29436c8713c254 based on LLVM d374a78545770f0325eca45a672b5b881798e87d revision)`) and it didn't work with json enabled. 
Is there may a way to add new extensions but still be backward compatible?

Comment (ClausKlein) [reply git-clang-format.py]: V17 and newer

File: git-clang-format.py [hunk 2]
```diff
@@ -79,14 +79,18 @@ def main():
       'c', 'h',  # C
       'm',  # ObjC
       'mm',  # ObjC++
-      'cc', 'cp', 'cpp', 'c++', 'cxx', 'hh', 'hpp', 'hxx', 'inc', 'inl',  # C++
-      'cu',  # CUDA
+      'cc', 'cp', 'cpp', 'c++', 'cxx', 'hh', 'hpp', 'hxx', 'inc',  # C++
+      'ccm', 'cppm', 'cxxm', 'c++m',  # C++ Modules
```
Comment (TheLartians) [inline git-clang-format.py:83]: Any reason to remove the `inl`  extension?

**Review by TheLartians — COMMENTED** [review]
Thanks for adding the new extensions! Seems there is still an issue with the CI being outdated. I'll investigate and update the PR.

File: git-clang-format.py (hunk 2 above)
Comment (ClausKlein) [reply git-clang-format.py:83]: Note: **asio** use `ipp`, **boost**, and **ACE** use `inl`

**Comment by TheLartians** [comment]
CI is now fixed, but there's a merge conflict, and I think we should keep the `inl` extension.

File: git-clang-format.py (hunk 2 above)
Comment (breyerml) [reply git-clang-format.py]: I've re-added the inl file ending and additionally added ipp.

**Comment by TheLartians** [comment]
Thanks for the update! I'm not sure why the CI is not running (and there seems to be no option for me to start it). Can you check if GitHub Actions is enabled on your branch?

**Comment by TheLartians** [comment]
Sorry, I found the issue now and could start the workflow manually! (I was using the "new merge experience" which seems to be missing the button)

**Review by TheLartians — APPROVED** [review]
Thanks for adding the extension!

**Comment by breyerml** [comment]
For some reason, at one point, the actual ".hip" extension was lost again in this PR :/

**Pull request merged**